src/databricks/labs/lsql/dashboards.py

```diff
@@ -851,39 +885,61 @@ def save_to_folder(self, dashboard: Dashboard, local_path: Path) -> Dashboard:
         dashboard = self._with_better_names(dashboard)
         for dataset in dashboard.datasets:
             query = QueryTile.format(dataset.query)
-            with (local_path / f"{dataset.name}.sql").open("w") as f:
-                f.write(query)
+            (local_path / f"{dataset.name}.sql").write_text(query)
         for page in dashboard.pages:
             with (local_path / f"{page.name}.yml").open("w") as f:
                 yaml.safe_dump(page.as_dict(), f)
+            for layout in page.layout:
+                if layout.widget.textbox_spec is not None:
+                    (local_path / f"{layout.widget.name}.md").write_text(layout.widget.textbox_spec)
         return dashboard
 
-    def deploy_dashboard(
+    def create_dashboard(
         self,
-        lakeview_dashboard: Dashboard,
+        dashboard_metadata: DashboardMetadata,
         *,
         parent_path: str | None = None,
         dashboard_id: str | None = None,
         warehouse_id: str | None = None,
     ) -> SDKDashboard:
-        """Deploy a lakeview dashboard."""
-        serialized_dashboard = json.dumps(lakeview_dashboard.as_dict())
-        display_name = lakeview_dashboard.pages[0].display_name or lakeview_dashboard.pages[0].name
+        """Create a Lakeview dashboard.
+
+        Parameters :
+            dashboard_metadata : DashboardMetadata
+                The dashboard metadata
+            parent_path : str | None (default: None)
+                The folder in the Databricks workspace to store the dashboard file in
+            dashboard_id : str | None (default: None)
+                The id of the dashboard to update
+            warehouse_id : str | None (default: None)
+                The id of the warehouse to use
+        """
+        dashboard_metadata.validate()
+        serialized_dashboard = json.dumps(dashboard_metadata.as_lakeview().as_dict())
         if dashboard_id is not None:
-            dashboard = self._ws.lakeview.update(
+            sdk_dashboard = self._ws.lakeview.update(
                 dashboard_id,
-                display_name=display_name,
+                display_name=dashboard_metadata.display_name,
                 serialized_dashboard=serialized_dashboard,
                 warehouse_id=warehouse_id,
             )
         else:
-            dashboard = self._ws.lakeview.create(
-                display_name,
+            sdk_dashboard = self._ws.lakeview.create(
+                dashboard_metadata.display_name,
                 parent_path=parent_path,
                 serialized_dashboard=serialized_dashboard,
                 warehouse_id=warehouse_id,
             )
-        return dashboard
+        return sdk_dashboard
+
+    def deploy_dashboard(self, dashboard: Dashboard, **kwargs) -> SDKDashboard:
+        """Legacy method use :meth:create_dashboard instead."""
+        warnings.warn("Deprecated method use `create_dashboard` instead.", category=DeprecationWarning)
```

Comment:
To use the `DashboardMetadata.validate` on the before dashboard creation, the signature is updated to expect `DashboardMetadata` instead of `Dashboard`. For backwards capability, the `deploy_dashboard` signature is unchanged, but intended to be deprecated. The implementation below creates the `DashboardMetadata` from the `Dashboard` through saving the dashboard to a temporary folder.

@nfx: Do you want to add a date to the deprecation warning?

Reply:
one transition version is fine